homeassistant/components/tplink/entity.py

```diff
@@ -292,7 +292,9 @@ def _description_for_feature[_D: EntityDescription](
                 translation_key=translation_key,
                 name=entity_name,  # if undefined will use translation key
                 entity_category=category,
-                entity_registry_enabled_default=enabled_default,
+                # enabled_default can be overridden to False in the description
+                entity_registry_enabled_default=enabled_default
+                and desc.entity_registry_enabled_default,
```

Comment:
This logic is a bit confusing. Maybe change `enabled_default` to `is_debug_feature` so it better shows intent

Reply:
That's done